DietPi-Software | Replace youtube-dl with yt-dlp

pulpe opened the pull request:
Based on discussion MichaIng/DietPi#6374

Should be ok to merge, tested on virtual dietpi.

pulpe committed:
Replace youtube-dl with yt-dlp

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -5647,15 +5647,14 @@ _EOF_
 
-			# youtube-dl supports Python 2 and Python 3, but its shebang calls Python 2, else fails: https://github.com/ytdl-org/youtube-dl/issues/27649
-			G_EXEC sed -i '1s/python$/python3/' /usr/local/bin/youtube-dl
+			aDEPS=('python3' 'ffmpeg')
```
Why is FFmpeg required? If so, then it should be installed as dependency on software ID 7 (which is FFmpeg).

pulpe replied:
As stated in [dependencies](https://github.com/yt-dlp/yt-dlp#dependencies). 

> While all the other dependencies are optional, ffmpeg and ffprobe are highly recommended
> 
> Required for [merging separate video and audio files](https://github.com/yt-dlp/yt-dlp#format-selection) as well as for various [post-processing](https://github.com/yt-dlp/yt-dlp#post-processing-options) tasks.

I'll update dependencies then.

pulpe committed:
Fix ffmpeg dependency of yt-dlp

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -5647,15 +5648,14 @@ _EOF_
 
-			# Remove obsolete /usr/local/bin/youtube-dl-py2 we used previously
-			[[ -f '/usr/local/bin/youtube-dl-py2' ]] && G_EXEC rm /usr/local/bin/youtube-dl-py2
+			G_EXEC ln -s /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```
Skip the removal before the download, so users as not left without any binary if the download fails for some reason. Instead use `-f` option to overwrite any existing file or symlink here:
```suggestion
			G_EXEC ln -sf /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```
The py2 removal could/should then stay where it was.

pulpe replied:
Done.

pulpe committed:
Modified by suggestions from MichaIng

MichaIng committed:
v8.18

- CHANGELOG | youtube-dl: Since the development of the original youtube-dl project stalled for some years, we migrated to the well known actively developed fork "yt-dlp": https://github.com/yt-dlp/yt-dlp. If you installed youtube-dl before, you can migrate via reinstall: "dietpi-software reinstall 195". For backwards compatibility, the "youtube-dl" command will stay valid as a symlink to "yt-dlp", but there are some differences you should be aware about when doing the migration on your system: https://github.com/yt-dlp/yt-dlp#differences-in-default-behavior. Many thanks to @rgabbo for suggestion and @pulpe for implementing this change: https://github.com/MichaIng/DietPi/discussions/5670, https://github.com/MichaIng/DietPi/pull/6380
